tensorflow/python/ops/ragged/ragged_embedding_ops.py

```diff
@@ -0,0 +1,242 @@
+      to be looked up in 'embedding_weights' of shape [r0, ..rN]. Values must be
+      in the range '[0, embedding_weights.shape[0]]'.
+    partition_strategy: A string specifying the partitioning strategy.
+    max_norm: If not `None`, each embedding is clipped if its l2-norm is larger
+      than this value.
+    name: A name for the operation (optional)
+
+  Returns:
+    A ragged tensor of shape [r0, r1, ...rN, e1, ...eM].
+
+  Raises:
+    ValueError: When params is empty or the type of the ids is not int32 or
+      int64.
+  """
+  if params is None:
+    raise ValueError("The embedding weights must be specified.")
+  if isinstance(params, (list, tuple)) and not params:
+    raise ValueError("The embedding weights should not be empty.")
+  if ids.dtype != dtypes.int32 and ids.dtype != dtypes.int64:
+    raise ValueError("The values contained by the inputs have type "
+                     f"{str(ids.dtype)}"
+                     " and cannot be processed. All values"
+                     " should be indices, either of type `int32` or `int64`.")
+
+  with ops.name_scope(name, "embedding_lookup_ragged") as name:
+    looked_up_ragged = ragged_functional_ops.map_flat_values(
+        embedding_ops.embedding_lookup,
+        params=params,
+        ids=ids,
+        partition_strategy=partition_strategy,
+        max_norm=max_norm)
+
+    return looked_up_ragged
+
+@dispatch.dispatch_for_api(embedding_ops.embedding_lookup_sparse)
+def embedding_lookup_sparse(params,
+                            sp_ids: ragged_tensor.Ragged,
+                            sp_weights,
+                            partition_strategy="mod",
+                            name=None,
+                            combiner=None,
+                            max_norm=None,
+                            allow_dense_grads=False):
+  if combiner is None:
+    combiner = "mean"
+  if combiner not in ("mean", "sqrtn", "sum"):
+    raise ValueError(
+        f"combiner must be one of 'mean', 'sqrtn' or 'sum', got {combiner}")
+  if isinstance(params, variables.PartitionedVariable):
+    params = list(params)  # Iterate to get the underlying Variables.
+  if not isinstance(params, list):
+    params = [params]
+  ignore_weights = sp_weights is None
+  if not ignore_weights:
+    if not isinstance(sp_weights, ragged_tensor.RaggedTensor):
+      raise TypeError(f"sp_ids must be of the same type as sp_weights.")
+    sp_ids.values.get_shape().assert_is_compatible_with(
+        sp_weights.values.get_shape())
+    sp_ids.get_shape().assert_is_compatible_with(
+        sp_weights.get_shape())
+    # TODO(yleon): Add enhanced node assertions to verify that sp_ids and
+    # sp_weights have equal indices and shapes.
+
+  with ops.name_scope(name, "embedding_lookup_sparse",
+                      params + [sp_ids]) as name:
+
+    segment_ids = sp_ids.value_rowids()
+    ids = sp_ids.flat_values
+
+    if len(params) == 1 and max_norm is None and allow_dense_grads:
```

Comment:
Could add some comments here to specify the upsides and downsides of going this branch?